Author: aszlig <aszlig@nix.build>
Date:   Thu May 3 06:29:02 2018 +0200

    Add support for systemd startup notification
    
    This is useful if we specifically want to wait for the habitica service
    during tests but might be useful for other things as well.
    
    Signed-off-by: aszlig <aszlig@nix.build>
    Filename: systemd-notify.patch

diff --git a/package.json b/package.json
index 1d5f379a88..66c75673b7 100644
--- a/package.json
+++ b/package.json
@@ -70,6 +70,7 @@
     "regenerator-runtime": "^0.13.3",
     "rimraf": "^2.4.3",
     "sass-loader": "^7.0.3",
+    "sd-notify": "^2.3.1",
     "shelljs": "^0.8.2",
     "superagent": "^5.0.2",
     "svg-inline-loader": "^0.8.0",
diff --git a/website/server/server.js b/website/server/server.js
index d704cdb1a1..03dc694f54 100644
--- a/website/server/server.js
+++ b/website/server/server.js
@@ -3,6 +3,7 @@ import logger from './libs/logger';
 import express from 'express';
 import http from 'http';
 import systemdSocket from 'systemd-socket';
+import notify from 'sd-notify';
 
 const server = http.createServer();
 const app = express();
@@ -29,6 +30,7 @@ attachMiddlewares(app, server);
 server.on('request', app);
 server.listen(systemdSocket() || app.get('port'), () => {
   logger.info(`Express server listening on port ${app.get('port')}`);
+  notify.ready();
 });
 
 module.exports = server;
